core/src/main/java/org/infinispan/commands/read/EntrySetCommand.java

```diff
@@ -57,6 +57,10 @@ public LoadType loadType() {
 
    @Override
    public Set<CacheEntry<K, V>> perform(InvocationContext ctx) throws Throwable {
+      Object lockOwner = ctx.getLockOwner();
+      if (ctx.getLockOwner() != null) {
+         return new BackingEntrySet<>(cache.getAdvancedCache().lockAs(lockOwner));
```

Comment:
When you start another locked stream holding the lock, the second iteration will go through the same entry. There, it won't be blocked (because the lock owners match), but when the inner iteration will leave the entry, it will unlock it, before the consumer in the outer iteration finished. Or am I mistaken? This sounds like it needs to be confirmed by a test :)